File: binary-tree.py
from random import randrange
from sys import setdlopenflags

# binary tree node class
class Node:
    def __init__(self, value) -> None:
        self.value = value
        self.left = None
        self.right = None

    def insert(self, value):
        if value < self.value:
            if self.left is None:
                self.left = Node(value)
            else:
                self.left.insert(value)
        elif value > self.value:
            if self.right is None:
                self.right = Node(value)
            else:
                self.right.insert(value)
    
    def inorder(self):
        if self.left is not None:
            self.left.inorder()
        print(self.value)
        if self.right is not None:
            self.right.inorder()

# initialize a tree of random integers from 1 to 100
nodes = [randrange(1,101) for i in range(11)]
tree = Node(nodes.pop())
for n in nodes:
    tree.insert(n)
tree.inorder()

File: binary-tree.js
class Node {
    constructor(value) {
        this.value = value;
        this.left = null;
        this.right = null;
    }

    insert(value) {
        if (value < this.value) {
            if (this.left === null) {
                this.left = new Node(value);
            } else {
                this.left.insert(value);
            }
        } else if (value > this.value) {
            if (this.right === null) {
                this.right = new Node(value);
            } else {
                this.right.insert(value)
            }
        }
    }

    inorder() {
        if (this.left !== null) {
            this.left.inorder();
        }
        console.log(this.value)
        if (this.right !== null) {
            this.right.inorder();
        }
    }
}

// initialize a tree of random integers from 1 to 100
const nodes = [...Array(11)].map(() => Math.ceil(Math.random() * 100));
const tree = new Node(nodes.pop());
for (n of nodes) {
    tree.insert(n)
}
tree.inorder()